File: backend/Models/Account.cs
namespace FinTrack.Api.Models;

public class Account
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = "bank"; // bank, cash, wallet
    public decimal Balance { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
}

File: src/types/account.ts
export interface Account {
  id: string;
  name: string;
  type: "bank" | "cash" | "wallet";
  balance: number;
  createdAt: string;
}

export interface AccountFormData {
  name: string;
  type: "bank" | "cash" | "wallet";
  balance: number;
}
